minimal markdownlint CLI

efx opened the pull request:
This was much easier to configure and it look's like we can preserve the shell runner script pattern.
By that I mean ./bin/markdown_lint.sh does the work. And a contributor can run it locally too.
Close #1132.

## questions
- can I lint hidden files (.meta/design.md)? yes, will need to modify invocation
- can I exclude legacy files? Yes
- can I start with a small subset of linting rules? yes. 
- what is the CLI? Using: https://github.com/igorshubovych/markdownlint-cli#markdownlint-cli

efx commented on bin/markdown_lint.sh:
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env sh
+npx markdownlint-cli concepts/**/*.md
```
TODO: find all v3 markdown files

coriolinus replied:
Probably we can get a lot closer just with

```suggestion
npx markdownlint-cli concepts/**/*.md exercises/**/*.md
```

coriolinus replied:
`set -e` please.

coriolinus reviewed (approved):
Seems very plausible. What would be very handy would be a demonstration: create a commit which violates one of the linting rules, verify that CI fails, revert it, verify that CI passes again. Still, I'm willing to approve on the basis of the text.

efx commented:
> create a commit which violates one of the linting rules, verify that CI fails, revert it, verify that CI passes again.

Great idea!

I introduced trailing whitespace which caused a [failure](https://github.com/exercism/rust/pull/1140/checks?check_run_id=1910450691#step:4:5).

Then, reverted in 144aeb9 which [passed](https://github.com/exercism/rust/pull/1140/checks?check_run_id=1910461247).
